wordpress-mobile/WordPress-Android
# Issue/remove nested comments when removing parent

This PR changes the way we handled orphaned comments in the way that is consistent with previous versions of the app.

We tried to show orphaned comments when the parent is removed, but this approach is not very compatible with paging, and causes more troubles than benefits.

After PR the orphaned comments will be hidden. Removing the parent comment will also remove nested comments.

To test:
- Navigate to reader comment list with nested comments and enough top level comments to trigger paging.
- Remove top level comment and confirm that the nested comment is removed as well.
- Reply to the post when not all the comments are loaded.
- Reply to comment.
- Smoke test general comment list behavior.

## Regression Notes
1. Potential unintended areas of impact
- Unforeseen issue with comment list. Nothing specific that I can think of.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Legacy code is not very testable.

PR submission checklist:
